packages/core/src/shared/sam/utils.ts

```diff
@@ -108,6 +109,38 @@ export async function updateRecentResponse(
         getLogger().warn(`sam: unable to save response at key "${key}": %s`, err)
     }
 }
+
+/**
+ * Returns true if there's an ongoing build process for the provided template, false otherwise
+ * @Param templatePath The path to the template.yaml file
+ */
+function isBuildInProgress(templatePath: string): boolean {
```

Comment:
seems like these new functions would fit in  `shared/sam/build.ts` ?

Reply:
Thanks. Updated.